### DISCUSSION TITLE
Scan only a list of files from command line

### DISCUSSION THREAD

**Discussion by nougatbyte** [navidrome/navidrome#2963] [Q&A] [answered]
Hi,

First of all: Love the software, thanks for all the effort <3

My Media is stored on a rather slow cloud storage which I mounted in my debian System.
Performance of navidrome is good but even a quick scan takes quite some time for me (big library, slow cloud).

Since most of the time I add music via scripts and know exactly which files where added, it would be great to have a way to scan only those files instead of checking the whole directory. I imagine something like

navidrome -c /path/to/conf scan --only=/mnt/cloud/track1.mp3,/mnt/cloud/track2.mp3

If this is not something you would implement, could you please point me roughly to the points in the code where I could implement it myself?

Thank you!

**Comment by deluan** [comment, marked as answer]
I'm in the process of rewriting the current scanner, which does not allow for easily accomplish what you are suggesting. 

The new one will allow specifying a folder to check for changes, and if your cloud storage mount works with [fsnotify](https://github.com/fsnotify/fsnotify), it will automatically do what you want: It will scan only changed folders
